doc/gold.rst

```diff
@@ -0,0 +1,412 @@
+
+Block-level usage: ::
+
+  let [mut] <identifier> : <type> = <expression>
+  let stash <identifier> : <type> = <known-expression>
+
+Binds a name to a value. In the first form, memory is allocated statically (i.e.
+memory is reserved, but the value is not included in the program image), and the
+value is computed and stored when the statement is executed. In the second form,
+memory is allocated in the program image with the initial value stored.
+
+By default, let-bindings are immutable, and may not be re-bound (though they may
+be shadowed). If the ``mut`` or ``stash`` storage classes are applied, then the
+binding becomes mutable, and the value may be changed.
+
+
+``fun``
+~~~~~~~
+
+Top-level usage: ::
+
+  fun <identifier>([<identifier> : <type> [, ...]]) [-> <type-expression>]
+    [...]
+  endfun
+
+Binds a name to a function with zero or more arguments and an optional return
+type. Introduces a new scope, and statements inside are considered block-level
+statements.
+
+A function with a return type must terminate by executing a ``return``
+statement.
+
+Note that the type of the binding introduced is a function type. Function types
+may only be used to call the function or get a pointer to its address using the
+``&`` operator.
+
+
+``isr``
+~~~~~~~
+
+Top-level usage: ::
+
+  isr <identifier>
+    [...]
+  endisr 
+
+Binds a name to an interrupt service routine. Introduces a new scope, and
+statements inside are considered block-level statements.
+
+Note that the type of the binding introduced is an ISR type. ISR types may only
+be used to get a pointer to its address using the ``&`` operator.
+
+
+Control Flow Statements
+-----------------------
+
+Control flow statements may only be used in block-level contexts. Additional
+restrictions may apply to individual statements, depending on context.
+Gold-syntax programs are executed statement-by-statement unless a control-flow
+statement is encountered.
+
+
+``return``
+~~~~~~~~~~
+
+Usage: ::
+
+  return [<expression>]
+
+Terminates execution of the current function, returning control to the caller,
+and possibly returning a value. This will cause any currently-executing loops to
+terminate.
+
+If the current function does not have a return type, then the expression is
+disallowed; if the current function does have a return type, then the expression
+is required, and must have a type assignable to the return type of the function.
+
+May also be used inside an ISR, in which case the expression is always
+disallowed.
+
+
+``if``
+~~~~~~
+
+Usage: ::
+
+  if <expression> then
+    [...]
+  [elseif <expression> then
+    [...]]
+  [elseif...]
+  [else
+    [...]]
+  end
+
+Causes at most one of the blocks provided to execute. Expressions are tested in
+order, and the first expression to evaluate to ``true`` causes the corresponding
+block to be executed. If none of the expressions evalute to ``true``, the block
+after the ``else`` is executed, if present. Once an expression which evaluates
```

Comment:
"the block after the ``else`` is executed" -> "the block beginning with ``else``"

minor ambiguity